python/mlc_chat/compiler_pass/fuse_add_norm.py

```diff
@@ -0,0 +1,275 @@
+                        v_ax1 = T.axis.spatial(T.int64(1), T.int64(0))
+                        v_ax2 = T.axis.spatial(
+                            T.int64(hidden_size), ax2 * T.int64(1024) + ax1_fused_1
+                        )
+                        var_T_add_intermediate[v_ax0, v_ax1, v_ax2] = add_local[
+                            v_ax2 // T.int64(1024)
+                        ]
+                with T.block("T_multiply_red_rf_init"):
+                    vax1_fused_1, v0 = T.axis.remap("SS", [ax1_fused_1, ax0_fused])
+                    T.reads()
+                    T.writes(T_multiply_red_rf_local[vax1_fused_1, v0, T.int64(0)])
+                    T_multiply_red_rf_local[vax1_fused_1, v0, T.int64(0)] = T.float32(0)
+                for ax1_fused_0, u in T.grid(T.int64(4), 1):
+                    with T.block("T_multiply_red_rf_update"):
+                        vax1_fused_1, v0, vax1_fused_0 = T.axis.remap(
+                            "SSR", [ax1_fused_1, ax0_fused, ax1_fused_0]
+                        )
+                        T_multiply_red_rf_local[
+                            vax1_fused_1, v0, T.int64(0)
+                        ] = T_multiply_red_rf_local[vax1_fused_1, v0, T.int64(0)] + T.Cast(
+                            "float32", add_local[vax1_fused_0]
+                        ) * T.Cast(
+                            "float32", add_local[vax1_fused_0]
+                        )
+            for ax1_fused in range(T.int64(1)):
+                for ax0 in T.thread_binding(T.int64(1024), thread="threadIdx.x"):
+                    with T.block("T_multiply_red"):
+                        vax1_fused_1, v0 = T.axis.remap("RS", [ax0, ax0_fused])
+                        T.reads(T_multiply_red_rf_local[vax1_fused_1, v0, T.int64(0)])
+                        T.writes(T_multiply_red_shared[v0, T.int64(0)])
+                        with T.init():
+                            T_multiply_red_shared[v0, T.int64(0)] = T.float32(0)
+                        T_multiply_red_shared[v0, T.int64(0)] = (
+                            T_multiply_red_shared[v0, T.int64(0)]
+                            + T_multiply_red_rf_local[vax1_fused_1, v0, T.int64(0)]
+                        )
+            for ax0_fused_0 in range(T.int64(4)):
+                for ax0_fused_1 in T.thread_binding(T.int64(1024), thread="threadIdx.x"):
+                    with T.block("T_cast_2"):
+                        v0 = T.axis.spatial(batch_size, ax0_fused)
+                        v1 = T.axis.spatial(
+                            T.int64(hidden_size), ax0_fused_0 * T.int64(1024) + ax0_fused_1
+                        )
+                        var_T_cast_intermediate[v0, T.int64(0), v1] = T.Cast(
+                            "float16",
+                            T.rsqrt(
+                                T_multiply_red_shared[v0, T.int64(0)] * T.float32(0.000244140625)
+                                + T.float32(9.9999999999999995e-07)
```

Comment:
we should avoid hardcoded numbers and use rms_norm_eps instead

Reply:
I added eps just now